vhpidirect/arrays/intvector/maindefined/caux.c

```diff
@@ -0,0 +1,9 @@
+#include "caux.h"
```

Comment:
Isn't this somehow redundant to `caux.c` in the parent dir? Sure, here the size is saved in a variable, instead of using a `define`. However, initializing `int arraySize = (sizeof(intArray)/sizeof(int));` should work for both cases. The same applies to `intArray`. Nothing prevents the subexample from overwriting the pointer. The hardcoded content would be useless, but I think it pays off because we want to show that users can use `main` optionally and without modification of helper C functions.

Reply:
sizeof(intArray) in the subexamples returns sizeof(int*), not sizeof(int[])... In order make it more closely match the original example I can make this array statically defined. then we start to lose out on the extensive main() entry point passing examples to ghdl for various reasons.

I think that may be for the better though. Let's simplify this to exactly the same as the intvector example, just with the values being printed out in main before the ghdl_main() call.

An we can make the extensive examples of the cosimulation in a different PR??

Reply:
> then we start to lose out on the extensive main() entry point passing examples to ghdl for various reasons.

I don't understand this sentence. Can you please reword it?

Reply:
Sure. I want to be extra clear so that you can tell me if I have any misunderstandings on the examples you want.

The sub example (maindefined) is to show how to use a custom entry point (main.c) to preprocess the VHPIDIRECT shared int[] (in caux.c). 
I think I made the subexample a little more complicated than it needs to be by asking for user input to defined the size of the int[]. 
- This meant that the variable had to be a malloc'd int*
- It also then strayed away from the parent example (intvector)
- It also is a little too much too soon because it starts begging the questions of all the ways to pass arguments to main() and even arguments to ghdl_main() through main(). **

I am proposing that we cut all of the discussion out by making the sub example much more similar to the parent example (exactly the same, except a main.c that wraps ghdl_main. main() does nothing, at most prints out "ghdl simulation\n*****************".)

We then 'lose out' on the 3 points above, but I think that they start to get extensive enough that a whole section could be written on them (at least have their own examples in a different section). Those examples can be linked to in this one's documentation....

Reply:
> I see your point. Let's keep this example as is, just splitting allocation and initialization. Then, we can add a shintvectors as done with shint.

I think that I am agreeing with your earlier response.

Reply:
> I think that I am agreeing with your earlier response.

Can you open a new PR with the status of this one when I did that comment? I believe that we can have that merged almost as is. We just want two subcases: set the size in C or set the size in VHDL. In both cases, the allocation is done in C. In the first one, it's hardcoded. In the second one, malloc is called once only, and a second helper function is used to fill it in C.

Then, we will have a better perspective about what to add here. I created #14, so this can be about the auto-reallocation logic that you have.

Reply:
Bear [this](https://github.com/umarcor/ghdl-cosim/pull/12#issuecomment-615934412) in mind. The question we want to answer is "How do I allocate in C while defining the size in one language only?". We need to provide two cases: hardcode the size in C (and pass it to VHDL) and harcode the size in VHDL (and pass it C).